projects/igniteui-angular/src/lib/progressbar/progressbar.component.ts

```diff
@@ -184,10 +184,6 @@ export abstract class BaseProgressDirective {
 
         this._internalState.newVal = Math.round(toValue(toPercent(this.value, maxNum), maxNum));
         this._value = this._internalState.oldVal = Math.round(toValue(this.valueInPercent, maxNum));
-        if (maxNum < this._value) {
```

Comment:
Setting the `max` runtime with a value lesser than current value would potentially enter this if. However, in such case, `this._value` will be recalculated wrongly to 0 on line 186. IMO we should place the `maxNum < this._value` check in the  the first if statement in the max setter.

Reply:
@hanastasov I have added some tests and they pass without this snippet. Have you tested that the logic doesn't work without it, or is it an assumption. Either something is wrong with the tests, or there's some other logic also normalizing the `_value` depending on the new `max`.

Reply:
I tested it and it seemed to lead the progressbar in wrong state. Looking at `Value should not exceed the max limit when operating with floating numbers` test now.

Reply:
OK, turns out what I mentioned is an edge case when max is set to 0. I have added a test that fails now and will add a fix in next commit.

Reply:
I've added a fix for the edge case I explained. When outside the edge case, the logic in `progressValue` normalizes the `this._value`, as you assumed.